common/lessonplan/generate.ts

```diff
@@ -0,0 +1,298 @@
+                                url: dataUrl,
+                            },
+                        });
+
+                        // // Generate a unique filename
+                        // const uniqueFilename = `${uuidv4()}${path.extname(file.originalname)}`;
+                        // const savePath = path.join('./', uniqueFilename);
+
+                        // // Write the resized image to a file
+                        // await fs.writeFile(savePath, resizedBuffer);
+
+                        logger.debug(`Image processed and converted to base64`);
+                        return null;
+                    }
+
+                    let loader;
+                    let docs;
+
+                    if (file.mimetype === 'application/pdf') {
+                        loader = new PDFLoader(blob, { splitPages: true });
+                        docs = await loader.load();
+                    } else if (file.mimetype === 'application/vnd.openxmlformats-officedocument.wordprocessingml.document') {
+                        loader = new DocxLoader(blob);
+                        docs = await loader.load();
+                    } else if (file.mimetype === 'application/vnd.openxmlformats-officedocument.presentationml.presentation') {
+                        loader = new PPTXLoader(blob);
+                        docs = await loader.load();
+                    } else {
+                        throw new Error(`Unsupported file type: ${file.mimetype}`);
+                    }
+
+                    // Add filename and page numbers (only for PDFs) to the content
+                    const content =
+                        `Filename: ${file.originalname}\n\n` +
+                        docs
+                            .map((doc, index) => (file.mimetype === 'application/pdf' ? `Page ${index + 1}:\n${doc.pageContent}` : doc.pageContent))
+                            .join('\n\n');
+
+                    // Check if the content is empty (excluding filename and page numbers)
+                    const contentWithoutMetadata = content
+                        .replace(`Filename: ${file.originalname}\n\n`, '')
+                        .replace(/Page \d+:\n/g, '')
+                        .trim();
+                    if (contentWithoutMetadata.length === 0) {
+                        emptyFiles.push(file.originalname);
+                        return null;
+                    }
+
+                    return {
+                        name: file.originalname,
+                        content: content,
+                        type: file.mimetype,
+                    };
+                } catch (error) {
+                    logger.error(`Error fetching or parsing file with UUID ${uuid}: ${error.message}`);
+                    throw error;
+                }
+            })
+        );
+
+        const validFiles = allFiles.filter((file): file is NonNullable<typeof file> => file !== null);
+
+        if (emptyFiles.length > 0) {
```

Comment:
Why not just `throw` instead of `return null` above? Same for the empty files ...